lib/dataset/data_array.cpp

```diff
@@ -102,6 +102,16 @@ bool operator!=(const DataArray &a, const DataArray &b) {
   return !operator==(a, b);
 }
 
+bool equals_nan(const DataArray &a, const DataArray &b) {
+  if (!equals_nan(a.coords(), b.coords()))
+    return false;
+  if (!equals_nan(a.masks(), b.masks()))
```

Comment:
Do we allow dtypes other than bool in masks? What would NaN mean in a mask?

Reply:
There is no limitation of what can be in a mask, but binary operations (or reductions) raise an exception since you cannot `OR` anything but bool.